backend/ibutsu_server/controllers/artifact_controller.py

```diff
@@ -129,6 +131,10 @@ def upload_artifact(body, token_info=None, user=None):
     """
     result_id = body.get("result_id") or body.get("resultId")
     run_id = body.get("run_id") or body.get("runId")
+    if not is_uuid(result_id):
```

Comment:
I can't remember what `is_uuid` does for None, but since either of those fields is optional, at least one will be None at any time, and this might throw an exception.

Reply:
looks like it raises `TypeError: one of the hex, bytes, bytes_le, fields, or int arguments must be given`.
I fixed this check, so it happens only if the value is given.